### PR TITLE
Move files from `shared` package to `search-ui` package

### PR DESCRIPTION
**Pull request by limitedmage**
Part of #34237

Move files related to search results to the `search-ui` package.

## Test plan

There should be no major code changes here, just import changes. All tests and CI should pass.

## App preview:

- [Web](https://sg-web-jp-searchuirefactor.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gsihbfhcpi.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/components/index.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
 export * from './CommitSearchResult'
+export * from './CommitSearchResultMatch'
+export * from './FileSearchResult'
+export * from './LastSyncedIcon'
+export * from './RepoFileLink'
 export * from './RepoSearchResult'
+export * from './ResultContainer'
+export * from './SearchResultStar'
+export * from './SyntaxHighlightedSearchQuery'
+
```
Comment (abeatrix) [inline client/search-ui/src/components/index.ts:11]:
```suggestion
import styles from './FileMatchChildren.module.scss'
export { styles as FileMatchChildrenStyles }
```

The VS Code extension is using the FileMatchChildren.module.scss [here](https://github.com/sourcegraph/sourcegraph/blob/1c7eb1f044d6acb5b54a9e759530224d03aa7fa4/client/vscode/src/webview/search-panel/alias/FileMatchChildren.tsx#L19), is it possible to export it as well?

**Review by abeatrix — CHANGES_REQUESTED** [review]
https://github.com/sourcegraph/sourcegraph/blob/1c7eb1f044d6acb5b54a9e759530224d03aa7fa4/client/vscode/src/webview/search-panel/alias/FileMatchChildren.tsx#L19

`import { displayRepoName } from '@sourcegraph/shared/src/components/RepoLink'` ?

I thinks other than the two issues I've brought up, the  rest of the changes should not break what we have for the VS Code extension, thanks!

**Comment by limitedmage** [comment]
@abeatrix  Thanks for the review! Addressed your comments.

**Review by vdavid — APPROVED** [review]
JetBrains works well!
Thanks for tidying up, it's a lot nicer now! <3

**Pull request merged**